Lib/test/datetimetester.py

```diff
@@ -3861,6 +3861,22 @@ def utcoffset(self, t):
                         expected = 1
                     self.assertEqual(got, expected)
 
+        # Test tz offset with microseconds
+        d0 = base.replace(minute=5, tzinfo=timezone(timedelta(microseconds=3)))
+        d1 = base.replace(
+            minute=5, tzinfo=timezone(timedelta(microseconds=456)))
+        d2 = base.replace(
+            minute=5, tzinfo=timezone(timedelta(microseconds=98764)))
+        for x in d0, d1, d2:
+            for y in d0, d1, d2:
+                for op in lt, le, gt, ge, eq, ne:
+                    got = op(x, y)
+                    expected = op(
+                        -x.utcoffset().microseconds,
```

Comment:
why do we need the minus sign here?

Reply:
Because the utc offset contributes to the time in a negative way, in other words, the greater the utc offset, the earlier the actual time.